resources/views/asset_maintenances/edit.blade.php

```diff
@@ -96,11 +96,23 @@
         </div>
 
         <!-- Asset Maintenance Cost -->
+        <?php
+        $currency_type=null;
+        if ($item->asset && $item->asset->location) {
+            $currency_type = $item->asset->location->currency;
+        }
+        ?>
         <div class="form-group {{ $errors->has('cost') ? ' has-error' : '' }}">
           <label for="cost" class="col-md-3 control-label">{{ trans('admin/asset_maintenances/form.cost') }}</label>
           <div class="col-md-2">
             <div class="input-group">
-              <span class="input-group-addon">{{ $snipeSettings->default_currency }}</span>
+              <span class="input-group-addon">
```

Comment:
Wouldn't it be cleaner to just say something like:

```
 @if ($item->asset && $item->asset->location)
   {{ $item->asset->location->currency }}
   @else
   {{ $snipeSettings->default_currency }}
@endif
```

and skip the currency_type bit?

Also, what happens if an asset has a location, but no currency listed? 

Should probably be:

```
 @if (($item->asset) && ($item->asset->location) && ($item->asset->location->currency!=''))
   {{ $item->asset->location->currency }}
   @else
   {{ $snipeSettings->default_currency }}
@endif
```